engine/api/workflow/workflow_run_results.go

```diff
@@ -389,3 +392,252 @@ func LoadRunResultsByRunIDAndType(ctx context.Context, db gorp.SqlExecutor, runI
+
+	var artifactManagerInteg *sdk.WorkflowProjectIntegration
+	for i := range wr.Workflow.Integrations {
+		if wr.Workflow.Integrations[i].ProjectIntegration.Model.ArtifactManager {
+			artifactManagerInteg = &wr.Workflow.Integrations[i]
+		}
+	}
+	if artifactManagerInteg == nil {
+		var err = sdk.Errorf("artifact manager integration is not found for workflow %s/%s", wr.Workflow.ProjectKey, wr.Workflow.Name)
+		log.ErrorWithStackTrace(ctx, err)
+		for i := range runResults {
+			result := runResults[i]
+			// If the result is not an artifact manager, we do nothing but we consider it as synchronized
+			dbResult := dbRunResult(result)
+			dbResult.DataSync = new(sdk.WorkflowRunResultSync)
+			dbResult.DataSync.Sync = false
+			dbResult.DataSync.Error = err.Error()
+			log.Debug(ctx, "updating run result %s", dbResult.ID)
+			if err := gorpmapping.Update(db, &dbResult); err != nil {
+				return err
+			}
+		}
+		if err := db.Commit(); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	log.Info(ctx, "artifact manager %q found for workflow run", artifactManagerInteg.ProjectIntegration.Name)
+
+	var (
+		rtName                = artifactManagerInteg.ProjectIntegration.Config[sdk.ArtifactoryConfigPlatform].Value
+		rtURL                 = artifactManagerInteg.ProjectIntegration.Config[sdk.ArtifactoryConfigURL].Value
+		buildInfoPrefix       = artifactManagerInteg.ProjectIntegration.Config[sdk.ArtifactoryConfigBuildInfoPrefix].Value
+		tokenName             = artifactManagerInteg.ProjectIntegration.Config[sdk.ArtifactoryConfigTokenName].Value
+		lowMaturitySuffix     = artifactManagerInteg.ProjectIntegration.Config[sdk.ArtifactoryConfigPromotionLowMaturity].Value
+		artifactoryProjectKey = artifactManagerInteg.ProjectIntegration.Config[sdk.ArtifactoryConfigProjectKey].Value
+	)
+
+	// Load the token from secrets
+	secrets, err := LoadDecryptSecrets(ctx, db, wr, wr.RootRun())
+	if err != nil {
+		return err
+	}
+
+	var rtToken string
+	for _, s := range secrets {
+		if s.Name == fmt.Sprintf("cds.integration.artifact_manager.%s", sdk.ArtifactoryConfigToken) {
+			rtToken = string(s.Value)
+			break
+		}
+	}
+	if rtToken == "" {
+		err := sdk.Errorf("unable to find artifact manager %q token", artifactManagerInteg.ProjectIntegration.Name)
+		log.ErrorWithStackTrace(ctx, err)
+		for i := range runResults {
+			result := runResults[i]
+			dbResult := dbRunResult(result)
+			dbResult.DataSync = new(sdk.WorkflowRunResultSync)
+			dbResult.DataSync.Sync = false
+			dbResult.DataSync.Error = err.Error()
+			log.Debug(ctx, "updating run result %s", dbResult.ID)
+			if err := gorpmapping.Update(db, &dbResult); err != nil {
+				return err
+			}
+		}
+		if err := db.Commit(); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	// Instanciate artifactory client
+	artifactClient, err := artifact_manager.NewClient(rtName, rtURL, rtToken)
+	if err != nil {
+		return err
+	}
+
+	version := fmt.Sprintf("%d", wr.Number)
```

Comment:
If it's for build version, you have to use cds.version
If it's the build.version: keep it empty like now maybe ?

Reply:
the code is
```
       version := fmt.Sprintf("%d", wr.Number)
	if wr.Version != nil {
		version = *wr.Version
	}
```